sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -646,8 +646,11 @@ class RocksDB(
           // is enabled.
           if (shouldForceSnapshot.get()) {
             uploadSnapshot()
-            changelogWriter = None
-            changelogWriter.foreach(_.abort())
+            try {
+              changelogWriter.foreach(_.abort())
+            } finally {
+              changelogWriter = None
+            }
```

Comment:
Seems like https://github.com/apache/spark/pull/48148 includes this fix

Reply:
I came up with this PR around the same time as https://github.com/apache/spark/pull/48148.

Since the fix is independent of the above PR, do you think this can go in first ?

Thanks

Reply:
cc @anishshri-db

Reply:
@tedyu - Actually we can't do the abort at all here. we need to ensure that the changelog file is written out. Otherwise it breaks reading the state in change-data-feed mode. Pls take a look at my fix - thx !

Reply:
we can maybe just fix the update on the error message tho as part of this PR. Will let @HeartSaVioR decide

Reply:
This was actually fixed via #48125 - the PR #48125 was up a couple days ahead on this.
@tedyu Thanks for your contribution and nice finding.